src/graphics/programlib/programlib_phong.js

```diff
@@ -277,7 +277,7 @@ pc.programlib.phong = {
             } else {
                 code += chunks.specularAaNonePS;
             }
-            code += this._addMap("specular", options, chunks, uvOffset);
+            code += this._addMap(options.metalnessWorkflow? "metalness" : "specular", options, chunks, uvOffset);

```

Comment:
The internal boolean is not meant to be used by the public. They don't really need to set it to true anyway without setting metalness or metalness map. Why not just hide it from the user? It's kinda like the 'blendType' property in scene_material.js which sets whatever booleans it needs internally in the property.